controller/types/types.go

```diff
@@ -243,6 +243,12 @@ func JobDownEvents(count int) map[JobState]int {
 	return map[JobState]int{JobStateDown: count}
 }
 
+const (
+	PartitionTypeBackground = "background"
+	PartitionTypeSystem     = "system"
+	PartitionTypeUser       = "user"
```

Comment:
Make these an explicit type (e.g. `type PartitionType string`) then we can't slip up and assign random strings to `NewJob.Partition`